packages/@unimodules/core/ios/UMCore/UMUtilities.m

```diff
@@ -206,6 +206,31 @@ + (NSString *)hexStringWithCGColor:(CGColorRef)color
   }
 }
 
+// taken from https://stackoverflow.com/questions/46192280/detect-if-the-device-is-iphone-x
++ (BOOL)doesDeviceHaveNotch{ // check if iphone X or newer
+  if([[UIDevice currentDevice] userInterfaceIdiom] == UIUserInterfaceIdiomPhone) {
+    switch ((int)[[UIScreen mainScreen] nativeBounds].size.height) {
+      case 1136:
```

Comment:
Such switch looks a bit weird, maybe grouping the cases would help (you can do sth like `case 1334, 1920:`), but I believe there is a better way to check if the device has a notch.

Reply:
There can be a problem when new Iphone comes out :/ But there is no tool available for explicitly checking if there is a notch.

Reply:
maybe we can use `[[[UIApplication sharedApplication] delegate] window].safeAreaInsets.top`? 🤔